source/gui/guiHelper.py

```diff
@@ -363,7 +362,20 @@ def addDialogDismissButtons(self, buttons, separated=False):
 		self.dialogDismissButtonsAdded = True
 		return buttons
 
+
+def safeAppExit():
+	def ensureMainLoopExited():
+		if wx.GetApp().GetMainLoop():
+			wx.GetApp().ExitMainLoop()

```

Comment:
Perhaps a log.warning or at very least log.debugWarning here stating that we are falling back to calling exitingMainLoop? I assume that this should hardly ever happen?

Reply:
I noticed this happening on all exits. I think this happens as a few of our frames overrides the exiting process to call this function.

Reply:
I've updated the comments to explain why this code should execute